src/app/check-in/check-in.component.ts

```diff
@@ -13,96 +13,94 @@ import { ThsDataService } from '../services/ths-data.service';
 })
 
 export class CheckInComponent {
-   public patientId: string = '';
-   public firstName: string = '';
-   public lastName: string = '';
-   public lastFourSS: string = '';
-   public authenticationFlag: boolean = true;
+  public patientId: string = '';
+  public firstName: string = '';
+  public lastName: string = '';
+  public email: string = '';
+  public authenticationFlag: boolean = true;
+  
+  constructor(private router: Router, private tsDataService: TsScreenerDataService, private tfiDataService: TfiDataService, private thsDataService: ThsDataService) {};
+  
+  public onClick() {
+    if(this.isCredentialsValid()) {
+      sessionStorage.clear();
+      this.tsDataService.clearHistory();
+      this.tfiDataService.clearHistory();
+      this.thsDataService.clearHistory();
+      Utilities.setSessionStorage('patient-id', this.patientId);
+      Utilities.setSessionStorage('first-name', this.firstName);
+      Utilities.setSessionStorage('last-name', this.lastName);
+      Utilities.setSessionStorage('email', this.email);
+      console.log(this.patientId);
+      console.log(this.firstName);
+      console.log(this.lastName);
+      console.log(this.email);
+      this.router.navigateByUrl('/appointments');
+      console.log('log in with ' + this.patientId);
+    } else {
+      this.authenticationFlag = false;
+      this.patientId = '';
+      console.log('failed log in ' + this.patientId);
+    }
+  }
 
-   constructor(private router: Router, private tsDataService: TsScreenerDataService, private tfiDataService: TfiDataService, private thsDataService: ThsDataService) {};
+  isCredentialsValid() : boolean {
+    return this.isEmailValid() && this.isNameValid() && this.isPatientIdValid();
```

Comment:
This logic will fail if the user doesn't provide an email or a first or last name. We should be able to ONLY provide the user ID and we should still pass all checks.